src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/ExtractToCodeBehindCodeActionResolver.cs

```diff
@@ -0,0 +1,231 @@
+                codeBehindPath = Path.Combine(
+                    Path.GetDirectoryName(path),
+                    $"{Path.GetFileNameWithoutExtension(path)}{identifier}{Path.GetExtension(path)}.cs");
+                n++;
+            } while (File.Exists(codeBehindPath));
+            return codeBehindPath;
+        }
+
+        /// <summary>
+        /// Determine all explicit and implicit using statements in the code
+        /// document using the intermediate node.
+        /// </summary>
+        /// <param name="razorCodeDocument">The code document to analyze.</param>
+        /// <returns>An enumerable of the qualified namespaces.</returns>
+        private static IEnumerable<string> FindUsings(RazorCodeDocument razorCodeDocument)
+        {
+            return razorCodeDocument
+                .GetDocumentIntermediateNode()
+                .FindDescendantNodes<UsingDirectiveIntermediateNode>()
+                .Select(n => n.Content);
+        }
+
+        /// <summary>
+        /// Generate a complete C# compilation unit containing a partial class
+        /// with the given name, body contents, and the namespace and all
+        /// usings from the existing code document.
+        /// </summary>
+        /// <param name="className">Name of the resultant partial class.</param>
+        /// <param name="contents">Class body contents.</param>
+        /// <param name="razorCodeDocument">Existing code document we're extracting from.</param>
+        /// <returns></returns>
+        private static string GenerateCodeBehindClass(string className, string contents, RazorCodeDocument razorCodeDocument)
+        {
+            var namespaceNode = (NamespaceDeclarationIntermediateNode)razorCodeDocument
+                .GetDocumentIntermediateNode()
+                .FindDescendantNodes<IntermediateNode>()
+                .FirstOrDefault(n => n is NamespaceDeclarationIntermediateNode);
+
+            var @class = CSharpSyntaxFactory
+                .ClassDeclaration(className)
+                .AddModifiers(CSharpSyntaxFactory.ParseToken("public"), CSharpSyntaxFactory.ParseToken("partial"));
+
+            var mock = (ClassDeclarationSyntax)CSharpSyntaxFactory.ParseMemberDeclaration($"class Class {contents}");
+            @class = @class.AddMembers(mock.Members.ToArray());
+
+            var @namespace = CSharpSyntaxFactory
+                .NamespaceDeclaration(CSharpSyntaxFactory.ParseName(namespaceNode.Content))
+                .AddMembers(@class);
+
+            var usings = FindUsings(razorCodeDocument)
+                .Select(u => CSharpSyntaxFactory.UsingDirective(CSharpSyntaxFactory.ParseName(u)))
+                .ToArray();
+            var compilationUnit = CSharpSyntaxFactory
+                .CompilationUnit()
+                .AddUsings(usings)
+                .AddMembers(@namespace);
+
+            return compilationUnit.NormalizeWhitespace().ToFullString();
```

Comment:
if you have acess to the roslyn workspace, you can format using the actual user settings.

Reply:
And sadly wedon't 😢